### DISCUSSION TITLE
Container update failed.

### DISCUSSION THREAD

**Discussion by RoccoRakete** [nextcloud/all-in-one#2804] [Questions] [answered]
1. update AIO 

### Expected behavior 

It should update correctly

### Host OS 

Debian

#### Nextcloud AIO version 

6.11

#### Current channel 

stable

#### Other valuable info 
Trying to fix docker.sock permissions internally...
Creating docker group internally with id 998
[0;92mInitial startup of Nextcloud All-in-One complete!
You should be able to open the Nextcloud AIO Interface now on port 8080 of this server!
E.g. https://internal.ip.of.this.server:8080

If your server has port 80 and 8443 open and you point a domain to your server, you can get a valid certificate automatically by opening the Nextcloud AIO Interface via:
https://your-domain-that-points-to-this-server.tld:8443[0m
{"level":"info","ts":1687106072.913803,"msg":"using provided configuration","config_file":"/Caddyfile","config_adapter":""}
{"level":"info","ts":1687106072.916314,"msg":"failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 2048 kiB, got: 416 kiB). See https://github.com/quic-go/quic-go/wiki/UDP-Receive-Buffer-Size for details."}
[18-Jun-2023 16:34:32] NOTICE: fpm is running, pid 116
[18-Jun-2023 16:34:32] NOTICE: ready to handle connections
NOTICE: PHP message: Slim Application Error
Type: GuzzleHttp\Exception\ServerException

0
Message: Server error: `POST http://localhost/v1.41/containers/nextcloud-aio-talk/start` resulted in a `500 Internal Server Error` response:
{&quot;message&quot;:&quot;driver failed programming external connectivity on endpoint nextcloud-aio-talk (881694b0bb4327f185d524f1c7f2 (truncated...)
File: /var/www/docker-aio/php/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php

3
Trace: #0 /var/www/docker-aio/php/vendor/guzzlehttp/guzzle/src/Middleware.php(72): GuzzleHttp\Exception\RequestException::create(Object(GuzzleHttp\Psr7\Request), Object(GuzzleHttp\Psr7\Response), NULL, Array, NULL)
#1 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(209): GuzzleHttp\Middleware::GuzzleHttp\{closure}(Object(GuzzleHttp\Psr7\Response))
#2 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(158): GuzzleHttp\Promise\Promise::callHandler(1, Object(GuzzleHttp\Psr7\Response), NULL)
#3 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/TaskQueue.php(52): GuzzleHttp\Promise\Promise::GuzzleHttp\Promise\{closure}()
#4 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(251): GuzzleHttp\Promise\TaskQueue->run(true)
#5 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(227): GuzzleHttp\Promise\Promise->invokeWaitFn()
#6 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(272): GuzzleHttp\Promise\Promise->waitIfPending()
#7 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(229): GuzzleHttp\Promise\Promise->invokeWaitList()
#8 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(69): GuzzleHttp\Promise\Promise->waitIfPending()
#9 /var/www/docker-aio/php/vendor/guzzlehttp/guzzle/src/Client.php(189): GuzzleHttp\Promise\Promise->wait()
#10 /var/www/docker-aio/php/vendor/guzzlehttp/guzzle/src/ClientTrait.php(95): GuzzleHttp\Client->request('POST', 'http://localhos...', Array)
#11 /var/www/docker-aio/php/src/Docker/DockerActionManager.php(182): GuzzleHttp\Client->post('http://localhos...')
#12 /var/www/docker-aio/php/src/Controller/DockerController.php(48): AIO\Docker\DockerActionManager->StartContainer(Object(AIO\Container\Container))
#13 /var/www/docker-aio/php/src/Controller/DockerController.php(33): AIO\Controller\DockerController->PerformRecursiveContainerStart('nextcloud-aio-t...', true)
#14 /var/www/docker-aio/php/src/Controller/DockerController.php(192): AIO\Controller\DockerController->PerformRecursiveContainerStart('nextcloud-aio-a...', true)
#15 /var/www/docker-aio/php/src/Controller/DockerController.php(173): AIO\Controller\DockerController->startTopContainer(true)
#16 /var/www/docker-aio/php/vendor/slim/slim/Slim/Handlers/Strategies/RequestResponse.php(38): AIO\Controller\DockerController->StartContainer(Object(GuzzleHttp\Psr7\ServerRequest), Object(GuzzleHttp\Psr7\Response), Array)
#17 /var/www/docker-aio/php/vendor/slim/slim/Slim/Routing/Route.php(358): Slim\Handlers\Strategies\RequestResponse->__invoke(Array, Object(GuzzleHttp\Psr7\ServerRequest), Object(GuzzleHttp\Psr7\Response), Array)
#18 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(65): Slim\Routing\Route->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#19 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(65): Slim\MiddlewareDispatcher->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#20 /var/www/docker-aio/php/vendor/slim/slim/Slim/Routing/Route.php(315): Slim\MiddlewareDispatcher->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#21 /var/www/docker-aio/php/vendor/slim/slim/Slim/Routing/RouteRunner.php(68): Slim\Routing\Route->run(Object(GuzzleHttp\Psr7\ServerRequest))
#22 /var/www/docker-aio/php/vendor/slim/csrf/src/Guard.php(476): Slim\Routing\RouteRunner->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#23 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(168): Slim\Csrf\Guard->process(Object(GuzzleHttp\Psr7\ServerRequest), Object(Slim\Routing\RouteRunner))
#24 /var/www/docker-aio/php/vendor/slim/twig-view/src/TwigMiddleware.php(115): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#25 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(121): Slim\Views\TwigMiddleware->process(Object(GuzzleHttp\Psr7\ServerRequest), Object(Psr\Http\Server\RequestHandlerInterface@anonymous))
#26 /var/www/docker-aio/php/src/Middleware/AuthMiddleware.php(38): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#27 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(269): AIO\Middleware\AuthMiddleware->__invoke(Object(GuzzleHttp\Psr7\ServerRequest), Object(Psr\Http\Server\RequestHandlerInterface@anonymous))
#28 /var/www/docker-aio/php/vendor/slim/slim/Slim/Middleware/ErrorMiddleware.php(76): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#29 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(121): Slim\Middleware\ErrorMiddleware->process(Object(GuzzleHttp\Psr7\ServerRequest), Object(Psr\Http\Server\RequestHandlerInterface@anonymous))
#30 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(65): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#31 /var/www/docker-aio/php/vendor/slim/slim/Slim/App.php(199): Slim\MiddlewareDispatcher->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#32 /var/www/docker-aio/php/vendor/slim/slim/Slim/App.php(183): Slim\App->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#33 /var/www/docker-aio/php/public/index.php(181): Slim\App->run()
#34 {main}

**Comment by szaimen** [comment]
Hi, can you post the outout of `sudo netstat -tulpn` here?

**Comment by RoccoRakete** [comment]
Sure

```
> martin@NextCloud:~$ sudo netstat -tulpn
Aktive Internetverbindungen (Nur Server)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 0.0.0.0:8080            0.0.0.0:*               LISTEN      837/docker-proxy    
tcp        0      0 10.7.2.65:3478          0.0.0.0:*               LISTEN      476/turnserver      
tcp        0      0 10.7.2.65:3478          0.0.0.0:*               LISTEN      476/turnserver      
tcp        0      0 127.0.0.1:3478          0.0.0.0:*               LISTEN      476/turnserver      
tcp        0      0 127.0.0.1:3478          0.0.0.0:*               LISTEN      476/turnserver      
tcp        0      0 10.7.2.65:3478          0.0.0.0:*               LISTEN      476/turnserver      
tcp        0      0 10.7.2.65:3478          0.0.0.0:*               LISTEN      476/turnserver      
tcp        0      0 127.0.0.1:3478          0.0.0.0:*               LISTEN      476/turnserver      
tcp        0      0 127.0.0.1:3478          0.0.0.0:*               LISTEN      476/turnserver      
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      490/sshd: /usr/sbin 
tcp        0      0 10.7.2.65:3479          0.0.0.0:*               LISTEN      476/turnserver      
tcp        0      0 10.7.2.65:3479          0.0.0.0:*               LISTEN      476/turnserver      
tcp        0      0 127.0.0.1:3479          0.0.0.0:*               LISTEN      476/turnserver      
tcp        0      0 127.0.0.1:3479          0.0.0.0:*               LISTEN      476/turnserver      
tcp        0      0 10.7.2.65:3479          0.0.0.0:*               LISTEN      476/turnserver      
tcp        0      0 10.7.2.65:3479          0.0.0.0:*               LISTEN      476/turnserver      
tcp        0      0 127.0.0.1:3479          0.0.0.0:*               LISTEN      476/turnserver      
tcp        0      0 127.0.0.1:3479          0.0.0.0:*               LISTEN      476/turnserver      
tcp        0      0 0.0.0.0:11000           0.0.0.0:*               LISTEN      1230/docker-proxy   
tcp6       0      0 :::8080                 :::*                    LISTEN      844/docker-proxy    
tcp6       0      0 :::22                   :::*                    LISTEN      490/sshd: /usr/sbin 
tcp6       0      0 ::1:3478                :::*                    LISTEN      476/turnserver      
tcp6       0      0 ::1:3478                :::*                    LISTEN      476/turnserver      
tcp6       0      0 ::1:3478                :::*                    LISTEN      476/turnserver      
tcp6       0      0 ::1:3478                :::*                    LISTEN      476/turnserver      
tcp6       0      0 ::1:3479                :::*                    LISTEN      476/turnserver      
tcp6       0      0 ::1:3479                :::*                    LISTEN      476/turnserver      
tcp6       0      0 ::1:3479                :::*                    LISTEN      476/turnserver      
tcp6       0      0 ::1:3479                :::*                    LISTEN      476/turnserver      
tcp6       0      0 :::11000                :::*                    LISTEN      1242/docker-proxy   
udp        0      0 10.7.2.65:3478          0.0.0.0:*                           476/turnserver      
udp        0      0 10.7.2.65:3478          0.0.0.0:*                           476/turnserver      
udp        0      0 10.7.2.65:3478          0.0.0.0:*                           476/turnserver      
udp        0      0 10.7.2.65:3478          0.0.0.0:*                           476/turnserver      
udp        0      0 127.0.0.1:3478          0.0.0.0:*                           476/turnserver      
udp        0      0 127.0.0.1:3478          0.0.0.0:*                           476/turnserver      
udp        0      0 127.0.0.1:3478          0.0.0.0:*                           476/turnserver      
udp        0      0 127.0.0.1:3478          0.0.0.0:*                           476/turnserver      
udp        0      0 10.7.2.65:3479          0.0.0.0:*                           476/turnserver      
udp        0      0 10.7.2.65:3479          0.0.0.0:*                           476/turnserver      
udp        0      0 10.7.2.65:3479          0.0.0.0:*                           476/turnserver      
udp        0      0 10.7.2.65:3479          0.0.0.0:*                           476/turnserver      
udp        0      0 127.0.0.1:3479          0.0.0.0:*                           476/turnserver      
udp        0      0 127.0.0.1:3479          0.0.0.0:*                           476/turnserver      
udp        0      0 127.0.0.1:3479          0.0.0.0:*                           476/turnserver      
udp        0      0 127.0.0.1:3479          0.0.0.0:*                           476/turnserver      
udp6       0      0 ::1:3478                :::*                                476/turnserver      
udp6       0      0 ::1:3478                :::*                                476/turnserver      
udp6       0      0 ::1:3478                :::*                                476/turnserver      
udp6       0      0 ::1:3478                :::*                                476/turnserver      
udp6       0      0 ::1:3479                :::*                                476/turnserver      
udp6       0      0 ::1:3479                :::*                                476/turnserver      
udp6       0      0 ::1:3479                :::*                                476/turnserver      
udp6       0      0 ::1:3479                :::*                                476/turnserver      
```

**Comment by szaimen** [comment, marked as answer]
Apparentl you did install a turnserver on the host which uses port 3478 as well. That is why the talk container fails to start. But you can adjust the talk port by so: https://github.com/nextcloud/all-in-one#how-to-adjust-the-talk-port

**Comment by RoccoRakete** [comment]
Okay thank you, I will try this as soon as I'm home. 

Is this somewhere mentoined? Did I overlooked this?

**Reply by szaimen** [reply]
Not sure what you are looking for? Maybe this? https://github.com/nextcloud/all-in-one#which-ports-are-mandatory-to-be-open-in-your-firewallrouter

**Reply by RoccoRakete** [reply]
nah I mean during the setup? obviously it was my fault :)

**Comment by RoccoRakete** [comment]
this doesn't seem to fix it 

new docker command: 

> sudo docker run \
--sig-proxy=false \
--name nextcloud-aio-mastercontainer \
--restart always \
--publish 8080:8080 \
--env APACHE_PORT=11000 \
--env APACHE_IP_BINDING=0.0.0.0 \
--env TALK_PORT=3479 \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume /var/run/docker.sock:/var/run/docker.sock:ro \
nextcloud/all-in-one:latest

**Reply by szaimen** [reply]
> TALK_PORT=3479

Indeed port 3479 is also already in use by the turnserver on the host:
![image](https://github.com/nextcloud/all-in-one/assets/42591237/e1f2bb37-90a9-414d-bbe9-64802d88dd1a)

**Comment by RoccoRakete** [comment]
Sorry I'm an idiot, sorry for wasting your time... 

Everything is working now.

**Comment by brendan-pike** [comment]
I've run into the problem where on my host I'm also running jitsi-meet which includes a turnserver using port 3479.  What was the solution you found @RoccoRakete I'm wondering if I can assign a different port to one or have a single turnserver support both applications, I'm on a beginner in these areas so advice is welcomed.

**Reply by szaimen** [reply]
Hi, see https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-adjust-the-talk-port

**Comment by brendan-pike** [comment]
Hey look at that, thank you!
